File: JSFIles/BinarySearch.js
const binarySearch = (arr , target) => {
    let left = 0;
    let right = arr.length;

    while(left <= right){
        let mid = Math.floor((left + right) / 2)
        
        if(arr[mid] === target){
            return `found at ${mid}`
        }
        else if(arr[mid] < target){
            left = mid + 1
        }
        else{
            right = mid - 1
        }
    }

    return "Not found"
}

const arr = [1, 2,3,4, 5, 6, 7,8];
const target = 7;
console.log(binarySearch(arr, target))

File: PYFiles/BinarySearch.py
def binarySearch(arr , target):
    left = 0
    right = len(arr)
    
    while(left <= right):
        mid = (left + right) // 2
        if arr[mid] == target:
            return "The target is at {F}".format(F=mid)
        elif arr[mid] < target:
            left = mid + 1
        else:
            right = mid - 1    
    return "Not found"     

arr = [1, 2,3,4, 5, 6, 7,8];
target = 7;
print(binarySearch(arr, target))   